ddtrace/internal/datadog/profiling/dd_wrapper/src/code_provenance.cpp

```diff
@@ -0,0 +1,178 @@
+#include "code_provenance.hpp"
+
+#include <memory>
+#include <mutex>
+#include <set>
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <vector>
+
+namespace Datadog {
+
+void
+Datadog::CodeProvenance::postfork_child()
+{
+    get_instance().mtx.~mutex();            // Destroy the mutex
+    new (&get_instance().mtx) std::mutex(); // Recreate the mutex
+    get_instance().reset();                 // Reset the state
+}
+
+void
+Datadog::CodeProvenance::set_enabled(bool enable)
+{
+    std::lock_guard<std::mutex> lock(mtx);
+    enabled = enable;
+}
```

Comment:
It's possible, but not necessary, to improve this by making `enabled` a std::atomic.  Since it's a bool, it'll use a lock-free implementation on all the platforms we support, which will diminish the need for a lock.

In all fairness, there's almost certainly no need to do even _that_ much, but I think there's value in making it clear that an interface is intended for concurrent access.

Again--not necessary, since I don't think this is in a performance-critical path.  Plus there is something to be said about using uniform primitives across the interface, so maybe it's not even worth it.

Reply:
Updated to use `atomic`